junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -303,21 +311,54 @@ private static String replaceUnderscores(String name) {
 	 * via the {@link IndicativeSentencesGeneration @IndicativeSentencesGeneration}
 	 * annotation.
 	 *
+	 * <p>If you do not want to rely on a display name generator for individual
+	 * sentence fragments, you can supply custom text for individual fragments
+	 * via the {@link SentenceFragment @SentenceFragment} annotation.
+	 *
 	 * @since 5.7
 	 */
 	@API(status = STABLE, since = "5.10")
 	class IndicativeSentences implements DisplayNameGenerator {
 
+		/**
+		 * {@code @SentenceFragment} is used to configure a custom sentence fragment
+		 * for a sentence generated by the {@link IndicativeSentences}
+		 * {@code DisplayNameGenerator}.
+		 *
+		 * <p>Note that {@link DisplayName @DisplayName} always takes precedence
+		 * over {@code @SentenceFragment}.
+		 *
+		 * @since 5.13
+		 */
+		@Target({ ElementType.TYPE, ElementType.METHOD })
+		@Retention(RetentionPolicy.RUNTIME)
+		@API(status = EXPERIMENTAL, since = "5.13")
+		public @interface SentenceFragment {
+
+			/**
+			 * Custom sentence fragment for the annotated class or method.
+			 *
+			 * @return a custom sentence fragment; never blank or consisting solely
+			 * of whitespace
+			 */
+			String value();
+
+		}
+
 		static final DisplayNameGenerator INSTANCE = new IndicativeSentences();
 
+		private static final Logger logger = LoggerFactory.getLogger(IndicativeSentences.class);
+
 		private static final Predicate<Class<?>> notIndicativeSentences = clazz -> clazz != IndicativeSentences.class;
 
 		public IndicativeSentences() {
 		}
 
 		@Override
 		public String generateDisplayNameForClass(Class<?> testClass) {
-			return getGeneratorFor(testClass, emptyList()).generateDisplayNameForClass(testClass);
+			String sentenceFragment = getSentenceFragment(testClass);
+			return (sentenceFragment != null ? sentenceFragment
+					: getGeneratorFor(testClass, emptyList()).generateDisplayNameForClass(testClass));
 		}
```

Comment:
It's always good to actually write tests, because... I had originally missed this use case. 😎